### PR TITLE
Fix stub resource loading on Android 9, 10

### PR DESCRIPTION
**Pull request by yujincheng08**
They can only load resources from zip files

### CODE REVIEW

**Comment by canyie** [comment]
#6274

**Comment by vvb2060** [comment]
Directory support was broken in Android 9 and 10.

https://cs.android.com/android/_/android/platform/frameworks/base/+/c07aa702703388747bd6e9b1091127e2736ffcd8

Add ResourcesProvider.loadFromDirectory

This API allows a directory to be loaded as if it was a zipped APK.
This is a substitute for the DirectoryAssetProvider API that currently does not work in the native layer.

File: stub/src/main/java/com/topjohnwu/magisk/DownloadActivity.java [hunk 1]
```diff
@@ -183,19 +185,17 @@ private void loadResources() throws Exception {
...
-
-            // addAssetPath requires a directory containing AndroidManifest.xml on Android 6.0
-            try (var stubApk = new ZipFile(getPackageCodePath());
-                 var manifest = new FileOutputStream(new File(dir, "AndroidManifest.xml"))) {
-                var stubManifest = stubApk.getInputStream(
-                        stubApk.getEntry("AndroidManifest.xml"));
-                APKInstall.transfer(stubManifest, manifest);
+            File res = new File(getCodeCacheDir(), "res.apk");
+            try (var out = new ZipOutputStream(new FileOutputStream(res))) {
+                // addAssetPath requires a directory containing AndroidManifest.xml on Android 6.0
```
Comment (canyie) [inline stub/src/main/java/com/topjohnwu/magisk/DownloadActivity.java:190]:
```suggestion
                // addAssetPath requires a zip containing AndroidManifest.xml on older Android versions
```

Comment (vvb2060) [inline stub/src/main/java/com/topjohnwu/magisk/DownloadActivity.java:190]:
```suggestion
                // Android 6 requires AndroidManifest.xml, Android 9-10 directory support is broken
```

**Review by canyie — APPROVED** [review]

**Pull request merged**